tools/crm_resource_runtime.c

```diff
@@ -655,6 +655,60 @@ cli_resource_delete(crm_ipc_t *crmd_channel, const char *host_uname,
     return rc;
 }
 
+int
+cli_resource_delete_failures(crm_ipc_t *crmd_channel, const char *host_uname,
+                    resource_t *rsc, const char *operation,
+                    const char *interval, pe_working_set_t *data_set)
+{
+    int rc = pcmk_ok;
+
+    if (rsc == NULL) {
+        return -ENXIO;
+
+    } else if (rsc->children) {
+        GListPtr lpc = NULL;
+
+        for (lpc = rsc->children; lpc != NULL; lpc = lpc->next) {
+            resource_t *child = (resource_t *) lpc->data;
+
+            rc = cli_resource_delete_failures(crmd_channel, host_uname, child, operation,
+                                              interval, data_set);
+            if(rc != pcmk_ok) {
+                return rc;
+            }
+        }
+        return pcmk_ok;
+    }
+
+    for (xmlNode *xml_op = __xml_first_child(data_set->failed); xml_op != NULL;
+         xml_op = __xml_next(xml_op)) {
+
+        const char *node = crm_element_value(xml_op, XML_ATTR_UNAME);
+        const char *task = crm_element_value(xml_op, XML_LRM_ATTR_TASK);
+        const char *task_interval = crm_element_value(xml_op, XML_LRM_ATTR_INTERVAL);
+        const char *resource_name = crm_element_value(xml_op, XML_LRM_ATTR_RSCID);
+
+        if(resource_name == NULL) {
+            continue;
+        } else if(host_uname && safe_str_neq(host_uname, node)) {
+            continue;
+        } else if(rsc->id && safe_str_neq(rsc->id, resource_name)) {
+            continue;
+        } else if(operation && safe_str_neq(operation, task)) {
+            continue;
+        } else if(interval && safe_str_neq(interval, task_interval)) {
+            continue;
+        }
+
+        crm_debug("Erasing %s failure for %s (%s detected) on %s",
+                  task, rsc->id, resource_name, node);
+        rc = cli_resource_delete(crmd_channel, node, rsc, task,
```

Comment:
We should do a pe_find_resource_with_flags() for resource_name before this.

Reply:
I went big. Can you review? 
https://github.com/ClusterLabs/pacemaker/pull/1396